_functions_onlyoffice.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash -eu
+
+# Don't load it several times
+set +u
+${_FUNCTIONS_ONLYOFFICE_LOADED:-false} && return
+set -u
+
+# if the script was started from the base directory, then the
+# expansion returns a period
+if test "${SCRIPT_DIR}" == "."; then
+  SCRIPT_DIR="$PWD"
+  # if the script was not called with an absolute path, then we need to add the
+  # current working directory to the relative path of the script
+elif test "${SCRIPT_DIR:0:1}" != "/"; then
+  SCRIPT_DIR="$PWD/${SCRIPT_DIR}"
+fi
+
+do_get_onlyoffice_settings() {
+  env_var DEPLOYMENT_ONLYOFFICE_CONTAINER_NAME "${INSTANCE_KEY}_onlyoffice"
+}
+
+#
+# Drops all Onlyoffice datas used by the instance.
+#
+do_drop_onlyoffice_data() {
+  echo_info "Dropping onlyoffice data ..."
+  echo_info "Drops Onlyoffice container ${DEPLOYMENT_ONLYOFFICE_CONTAINER_NAME} ..."
+  delete_docker_container ${DEPLOYMENT_ONLYOFFICE_CONTAINER_NAME}
+  echo_info "Drops Onlyoffice docker volume ${DEPLOYMENT_ONLYOFFICE_CONTAINER_NAME} ..."
+  delete_docker_volume ${DEPLOYMENT_ONLYOFFICE_CONTAINER_NAME}
+  echo_info "Done."
+  echo_info "Onlyoffice data dropped"
+}
+
+do_create_onlyoffice() {
+  if ${DEPLOYMENT_ONLYOFFICE_DOCUMENTSERVER_ENABLED}; then
+    echo_info "Creation of the OnlyOffice Docker volume ${DEPLOYMENT_ONLYOFFICE_CONTAINER_NAME} ..."
+    create_docker_volume ${DEPLOYMENT_ONLYOFFICE_CONTAINER_NAME}
+    echo_info "OnlyOffice Docker volume ${DEPLOYMENT_ONLYOFFICE_CONTAINER_NAME} created"
+  fi
+}
+
+do_stop_onlyoffice() {
+  echo_info "Stopping OnlyOffice ..."
+
+  ensure_docker_container_stopped ${DEPLOYMENT_ONLYOFFICE_CONTAINER_NAME}
+  echo_info "OnlyOffice container ${DEPLOYMENT_ONLYOFFICE_CONTAINER_NAME} stopped."
+}
+
+do_start_onlyoffice() {
+  echo_info "Starting OnlyOffice..." 
+  echo_info "Starting OnlyOffice container ${DEPLOYMENT_ONLYOFFICE_CONTAINER_NAME} based on image ${DEPLOYMENT_ONLYOFFICE_IMAGE}:${DEPLOYMENT_ONLYOFFICE_IMAGE_VERSION}"
+
+  # Ensure there is no container with the same name
+  delete_docker_container ${DEPLOYMENT_ONLYOFFICE_CONTAINER_NAME}
+
+  ${DOCKER_CMD} run \
+    -d \
+    -p "127.0.0.1:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}:80" \
+    -v ${DEPLOYMENT_ONLYOFFICE_CONTAINER_NAME}:/var/log/onlyoffice  \
```

Comment:
Is the log the only important thing to keep between 2 restarts ?

Reply:
we can also keep the data folder : /var/www/onlyoffice/Data
According the ACC-111 chat in Tribe, with Gregory, this is useless since we won't store anything in documentserver, documentserver will download the original office file form eXo, and will push it back once saved changes.